typewriter/fixes/fixer_utils.py

```diff
@@ -133,3 +140,254 @@ def touch_import(package, name, node):
+        elif node.type == token.NAME and node.value.strip() == name:
+            return node
+    return None
+
+
+ImportPairing = NamedTuple("ImportPairing", [
+    ("entry", str),
+    ("binding_name", str)
+])
+
+ImportInfo = NamedTuple("ImportInfo", [
+    ("imports", Optional[Set[ImportPairing]]),
+    ("package", Optional[str]),
+    ("node", Node)
+])
+
+_def_syms = {syms.classdef, syms.funcdef}
+
+
+def find_binding(name, node, package=None):
+    """ Returns the node which binds variable name, otherwise None.
+        If optional argument package is supplied, only imports will
+        be returned.
+        See test cases for examples."""
+    for child in node.children:
+        ret = None
+        if child.type == syms.for_stmt:
+            if _find(name, child.children[1]):
+                return child
+            n = find_binding(name, make_suite(child.children[-1]), package)
+            if n:
+                ret = n
+        elif child.type in (syms.if_stmt, syms.while_stmt):
+            n = find_binding(name, make_suite(child.children[-1]), package)
+            if n:
+                ret = n
+        elif child.type == syms.try_stmt:
+            n = find_binding(name, make_suite(child.children[2]), package)
+            if n:
+                ret = n
+            else:
+                for i, kid in enumerate(child.children[3:]):
+                    if kid.type == token.COLON and kid.value == ":":
+                        # i+3 is the colon, i+4 is the suite
+                        n = find_binding(name, make_suite(child.children[i+4]), package)
+                        if n:
+                            ret = n
+        elif child.type in _def_syms and child.children[1].value == name:
+            ret = child
+        elif _is_import(child):
+            res = get_import_info(child)
+            if res.binding_name == name and res.package == package:
+                ret = res.node
+        elif child.type == syms.simple_stmt:
+            ret = find_binding(name, child, package)
+        elif child.type == syms.expr_stmt:
+            if _find(name, child.children[0]):
+                ret = child
+
+        if ret:
+            if not package:
+                return ret
+            if is_import(ret):
+                return ret
+    return None
+
+
+def find_import_info(package, name, node):
+    # type: (Optional[str], str, Node) -> Optional[ImportInfo]
+    """
+    Finds the import statement for <name> regardless of whether <name> is the binding name
+        ex where name='a':
+        import a => match AND import a as b => match BUT import b as a => None
+
+    Parameters
+    -----------
+    package : Optional[str]
+    name : str
+    node : Node
+
+    Return
+    -----------
+    Optional[ImportInfo]
+    """
+    for child in node.children:
+        if is_import(child):
+            ret = get_import_info(child)
+            if ret.package == package and dict(ret.imports).get(name) is not None:
```

Comment:
The `dict(...).get` works, but feels a bit roundabout to me - at minimum, you can do `name in dict(ret.imports)`, but something like `any(x.entry == name for x in ret.imports)`, though slightly wordier, feels more direct (and should actually be faster, since it's not bothering with hashing + building out a dict structure).

Alternatively - you could filter to only the bindings we care about, check if non-empty, and if so, use `return ret._replace(imports=filtered)` - that way, we'd be able to skip the `if pair.entry == name` check in `type_by_import_stmt`.